src/logic/fetch-elements/fetch-name-owner.ts

```diff
@@ -0,0 +1,30 @@
+import { AppComponents } from '../../types'
+
+const QUERY_NAME_OWNER: string = `
+  query fetchNameOwner($name: String) {
+    nfts(
+      where: {name_starts_with_nocase: $name, name_ends_with_nocase: $name }
+    ) {
+      owner {
+        address
+      }
+    }
+}`
+
+export type NameOwnerFromQuery = {
+  nfts: {
+    owner: {
+      address: string
+    }
+  }[]
+}
+
+export async function fetchNameOwner(
+  components: Pick<AppComponents, 'theGraph'>,
+  name: string
+): Promise<{
+  owner: string
+}> {
+  const result = await components.theGraph.ensSubgraph.query<NameOwnerFromQuery>(QUERY_NAME_OWNER, { name })
+  return { owner: result.nfts[0].owner.address }
```

Comment:
Searching for a name might return an empty `nfts` array due to a name not existing, which will make this function fail. Would you mind changing the function to take that into consdieration?